test/test_job.py

```diff
@@ -17,7 +17,7 @@
 
 
 class JobTests(unittest.TestCase):
-    def setUp(self):
+    def setUp(self) -> None:
```

Comment:
do we really want to annotate all test cases? seems a bit redundant/noisy...

Reply:
Adding type annotations to all of the test cases like that enables mypy to inspect the body of the test. From [urllib3's notes on adding type hints](https://sethmlarson.dev/blog/2021-10-18/tests-arent-enough-case-study-after-adding-types-to-urllib3):

> Once we completed the addition of type hints to the source code of urllib3 we set our sights on the test suite. This may seem strange as our users are very unlikely to directly benefit from us adding type hints there, however there’s one big benefit: we’re able to find issues with types from a user perspective!

Reply:
Nice! Didn't know about this, yet. Sounds good to me